### PR TITLE
[Examiners] Add Case Sensitive for the table.

### PR DESCRIPTION
**Pull request by kongtiaowang**
This pull request Add Case Sensitive for the table.
https://redmine.cbrain.mcgill.ca/issues/14728

### CODE REVIEW

**Comment by ZainVirani** [comment]
I'll test this in like 30 minutes

**Comment by driusan** [comment]
I don't understand why this one column in one table needs to be case sensitive if that's not a standard elsewhere in LORIS

**Comment by ZainVirani** [comment]
@driusan The issue is that this table is based on user Full Names instead of unique userIDs. So if I create 3 users with full names "Dave M" "dave m" and "DAVE M", and all of them are DCC examiners, only 1 entry will show up in this table ("dave m"). This PR only fixes the case sensitivity, but ideally this table would be based on userID (however this has its own issue, the add examiner section on the top right would need adjusting as well).

File: SQL/0000-00-00-schema.sql [hunk 1]
```diff
@@ -1157,7 +1157,7 @@ CREATE TABLE `family` (
 
 CREATE TABLE `examiners` (
   `examinerID` int(10) unsigned NOT NULL auto_increment,
-  `full_name` varchar(255) default NULL,
+  `full_name` varchar(255) CHARACTER SET utf8 COLLATE utf8_bin default NULL,
```
Comment (PapillonMcGill) [inline SQL/0000-00-00-schema.sql]: char set should be utf8mb4

Comment (driusan) [reply SQL/0000-00-00-schema.sql]: If case sensitivity is the issue, I think utf8mb4 is case insensitive.

**Comment by driusan** [comment]
@ZainVirani But isn't it a good thing that all of those would show up as one user? It would likely get confusing if "Dave M" and "dave m" were different users.

File: SQL/0000-00-00-schema.sql (hunk 1 above)
Comment (PapillonMcGill) [reply SQL/0000-00-00-schema.sql]: Some case sensitive collation now exist but maybe not for the version we use and maybe not with the proper ordering.

**Comment by driusan** [comment]
Ping.

Still not seeing why this is required on a one column basis or why we would want to make names, in particular, case sensitive.

**Comment by ZainVirani** [comment]
>But isn't it a good thing that all of those would show up as one user? It would likely get confusing if "Dave M" and "dave m" were different users.

What if they are different users? Keep in mind that we're talking about user full names here, not user names

**Comment by driusan** [comment]
@ZainVirani I think assuming names are case sensitive and treating them as separate users opens a can of worms with regards to security, where someone can easily impersonate someone else by creating a user of the same name with a different case

**Comment by ZainVirani** [comment]
@driusan that's exactly what the problem is now... again, if you as DaveM were to be listed as an examiner for site A, another user "davem" was an examiner at site B, davem would now be listed as an examiner at site A and B

**Comment by ridz1208** [comment]
I see your point @ZainVirani but I agree with @driusan.

you are not mitigating anything by making it case sensitive you are just adding complications usually. the code will need to check for casing.. then on the front end you can have a Dave M, DAVE m, DaVe m, dave M....

yes it is a problem that we are reliying solely on full name but the issue is not resolved by looking at the casing of the name but rather by appending an ID that makes each entry unique. It's a valid ticket for an improvement not a bug, its like this by design (or lack thereof) but this PR is not a solution

**Comment by zaliqarosli** [comment]
@ridz1208 @driusan are we saying we should close this PR? @ZainVirani

**Pull request closed without merging**